code/datums/autocells/explosion.dm

```diff
@@ -209,40 +189,13 @@
 
 			// Set the direction the explosion is traveling in
 			E.direction = dir
-			//Diagonal cells have a small delay when branching off the center. This helps the explosion look circular
-			if(!direction && (dir in GLOB.diagonals))
-				E.delay = 1
 
 			setup_new_cell(E)
+			new_cells += E
 
 	// We've done our duty, now die pls
 	qdel(src)
-
-/*
-The issue is that between the cell being birthed and the cell processing,
-someone could potentially move through the cell unharmed.
-
-To prevent that, we track all atoms that enter the explosion cell's turf
-and blow them up immediately once they do.
-
-When the cell processes, we simply don't blow up atoms that were tracked
-as having entered the turf.
-*/
-
-/datum/automata_cell/explosion/proc/on_turf_entered(atom/movable/A)
-	// Once is enough
-	if(A in exploded_atoms)
-		return
-
-	exploded_atoms += A
-
-	// Note that we don't want to make it a directed ex_act because
-	// it could toss them back and make them get hit by the explosion again
-	if(A.gc_destroyed)
-		return
-
-	INVOKE_ASYNC(A, TYPE_PROC_REF(/atom, ex_act), power, null, explosion_cause_data)
-	log_explosion(A, src)
```

Comment:
Where did this go? It's dumb but it's important, because that's how game ensures you can't just walk through an explosion, and that chained explosions work properly

Reply:
not needed anymore, since explosions are sped up so much, this actually makes the performance worse.

Reply:
It's here for a reason. It's not just about things moving normally into each other at walkspeed, it also is relevant for things like recursive throws into explosions. Also, even if explosions happened in perfect sync with the rest, you'd still need to scan turfs twice per tick to ensure you sample player positions correctly to pick them up during subsytem fire. Otherwise someone can walk through an explosion before you scan for them if they moved in another way during previous tick. Think of it like Shannon-Nyquist theorem

Reply:
well then it clearly doesn't work anyways considering that blast walking is still a thing

Reply:
Its also not relevant for throws